Apsitvarkom.DataAccess/PollutedLocationDTODatabaseRepository.cs

```diff
@@ -1,29 +1,47 @@
 ﻿using Apsitvarkom.Models;
 using Apsitvarkom.Models.DTO;
+using Apsitvarkom.Models.Mapping;
 using AutoMapper;
 using Microsoft.EntityFrameworkCore;
 
 namespace Apsitvarkom.DataAccess;
 
+/// <summary>
+/// Class for <see cref="PollutedLocationDTO" /> data handling using database.
+/// </summary>
 public class PollutedLocationDTODatabaseRepository : ILocationDTORepository<PollutedLocationDTO>
 {
-    private readonly PollutedLocationContext _context;
+    private readonly IPollutedLocationContext _context;
     private readonly IMapper _mapper;
 
-    public PollutedLocationDTODatabaseRepository(PollutedLocationContext context, IMapper mapper)
+    /// <summary>Constructor for the reader.</summary>
+    /// <param name="context">Database context for <see cref="PollutedLocation"/>.</param>
+    /// <param name="mapper">Mapper implementing profile <see cref="PollutedLocationProfile"/>.</param>
+    public PollutedLocationDTODatabaseRepository(IPollutedLocationContext context, IMapper mapper)
     {
         _context = context;
         _mapper = mapper;
     }
 
+    /// <inheritdoc />
     public async Task<IEnumerable<PollutedLocationDTO>> GetAllAsync()
     {
         var models = await _context.PollutedLocations.ToListAsync();
-
         return _mapper.Map<List<PollutedLocation>, IEnumerable<PollutedLocationDTO>>(models);
     }
 
-    public Task<IEnumerable<PollutedLocationDTO>> GetAllAsync(Location inRelationTo) => throw new NotImplementedException();
+    /// <inheritdoc />
+    public async Task<IEnumerable<PollutedLocationDTO>> GetAllAsync(Location inRelationTo)
+    {
+        return from pollutedLocation in await _context.PollutedLocations.ToListAsync()
+            orderby inRelationTo.DistanceTo(pollutedLocation)
+            select _mapper.Map<PollutedLocation, PollutedLocationDTO>(pollutedLocation);
+    }
```

Comment:
Same in this method as the previous comment, utilize `IQueryable` because right now you're just executing SQL query by calling `ToListAsync`. Preferably you would construct the query and then call `ToListAsync` at the end

Reply:
as we are using a DistanceTo method here, a simple SQL query to `order by` cannot resolve our method and we first have to get all the instances to list or to enumerable.
More information [here](https://learn.microsoft.com/en-us/ef/core/querying/client-eval).

![image](https://user-images.githubusercontent.com/86420873/201608808-8f8c86b0-72f0-43ad-ab5f-34d029a04750.png)

Reply:
yeah that makes sense since we have C# code in that order by. I guess C# LINQ and SQL worlds clash here and there isn't a nice way (at least how we got things set up right now). We could investigate this in the future, but good enough for now I guess